Assets/Scripts/Traffic System/CarCollision.cs

```diff
@@ -0,0 +1,42 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CarCollision : MonoBehaviour
+{
+    [SerializeField] GraphMovement car;
+    float carSpeed;
+    [SerializeField] float visionAngle; //radians
+    
+    private void Start() {
+        carSpeed = car.speed;
+        }
+    private void OnTriggerEnter(Collider other) {
+        Vector3 agentToVVertex = car.transform.position - other.gameObject.transform.position;
+        agentToVVertex.Normalize();
+        if (Vector3.Dot(agentToVVertex, car.transform.forward) > Mathf.Cos(visionAngle)) {
```

Comment:
change to if (!Vector3.Dot(agentToVVertex, car.transform.forward) > Mathf.Cos(visionAngle)) { return; } to limit nested ifs